Source/SelfService/Web/microservice/microserviceView/configuration/environmentVariablesSection/environmentVariableSection.tsx

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import React, { useEffect, useState } from 'react';
+
+import { useSnackbar } from 'notistack';
+
+import { GridColDef, GridRowId, GridRowModesModel, GridRowModes, GridRowModel } from '@mui/x-data-grid-pro';
+
+import { Box } from '@mui/material';
+import { AddCircle, DeleteRounded, DownloadRounded, ArrowDropDown } from '@mui/icons-material';
+
+import { Accordion } from '@dolittle/design-system/atoms/Accordion/Accordion';
+import { Button } from '@dolittle/design-system/atoms/Button';
+import { DataTablePro } from '@dolittle/design-system/atoms/DataTablePro/DataTablePro';
+
+import { getEnvironmentVariables, getServerUrlPrefix, InputEnvironmentVariable, updateEnvironmentVariables } from '../../../../api/api';
+
+import { RestartInfoBox } from '../restartInfoBox';
+import { EmptyDataTable } from '../emptyDataTable';
+
+interface EnvironmentVariableTableRow extends InputEnvironmentVariable {
+    id: GridRowId;
+    isNew: boolean;
+};
+
+type EnvironmentVariablesProps = {
+    applicationId: string;
+    environment: string;
+    microserviceName: string;
+    microserviceId: string;
+};
+
+export const EnvironmentVariablesSection = ({ applicationId, environment, microserviceName, microserviceId }: EnvironmentVariablesProps) => {
+    const { enqueueSnackbar } = useSnackbar();
+
+    const [envVariableTableRows, setEnvVariableTableRows] = useState<EnvironmentVariableTableRow[]>([]);
+    const [selectedRowIds, setSelectedRowIds] = useState<GridRowModel[]>([]);
+    const [rowMode, setRowMode] = useState<GridRowModesModel>({});
+    const [disableAddButton, setDisableAddButton] = useState(false);
+    const [restartInfoBoxIsOpen, setRestartInfoBoxIsOpen] = useState(false);
+
+    const columns: GridColDef<EnvironmentVariableTableRow>[] = [
+        {
+            field: 'name',
+            headerName: 'Name',
+            width: 330,
+            editable: true
+        },
+        {
+            field: 'value',
+            headerName: 'Value',
+            width: 330,
+            editable: true
+        },
+        {
+            field: 'isSecret',
+            headerName: 'Secret',
+            type: 'singleSelect',
+            valueOptions: [{ value: true, label: 'Yes' }, { value: false, label: 'No' }],
+            editable: true,
+            renderCell: ({ value }) => (
+                // TODO: Needs improvement
```

Comment:
What kind of improvement is needed? Is it something to create a follow-up task for, or could we look at this together, perhaps?

Reply:
Just styling.

1. In 'edit' mode, the select cell took too much space and overall row got too high.
2. After exiting from 'edit' mode, button from next line got focus.
3. Button had some ugly ripple effects and was wrong size.